### DISCUSSION TITLE
call go from v

### DISCUSSION THREAD

**Discussion by sh4m2hwz** [vlang/v#12683] [Ideas]
vlang has a C backend, here you can make stubs or intermediate C code that pulls go code, or compile go as c-archive or c-shared and thus pull go code, what are the alternative, faster methods of working with but code,  is there a porting of th code to v code?
P.S:
quite a lot of useful libraries and software are written in go

**Comment by JalonSolov** [comment]
Porting Go to V isn't hard to do manually for small routines, but can become quite complex for larger code bases.

There has been talk of a `go2v`, but nothing much has happened with that as yet.

Other than that, the way to interface between Go and V is to use `cgo` on the Go side, and mark things with `[export]` on the V side so everything can align between the 2.  I don't see that changing any time soon.

**Comment by sh4m2hwz** [comment]
Yes 
V code <-(C/CGO)-> Go code
